nexus/reconfigurator/planning/src/blueprint_builder/external_networking.rs

```diff
@@ -99,6 +109,13 @@ impl<'a> BuilderExternalNetworking<'a> {
             HashSet::new();
         let mut existing_boundary_ntp_v6_ips: HashSet<Ipv6Addr> =
             HashSet::new();
+        let mut existing_external_dns_v4_ips: HashSet<Ipv4Addr> =
+            HashSet::new();
+        let mut existing_external_dns_v6_ips: HashSet<Ipv6Addr> =
+            HashSet::new();
+        let mut available_external_dns_ips: BTreeSet<
+            OmicronZoneExternalFloatingAddr,
```

Comment:
I *think* we only use the IP from this, not the port or id.  Should we be storing just `IpAddr` here?

Reply:
We definitely could, and that's how I coded it initially. But then I noticed that I was converting it back into an `OmicronZoneExternalFloatingAddr` to put in the `blueprint_zone_type::ExternalDns::dns_address` field. So I'm going to answer your question with a question (sorry): is it better to create new floating address records, or to try to reuse them (copying the ID) for ... efficiency? Tracability? As I ask the question I realize what the answer probably should be, but I suppose I will ask it anyway.

Reply:
Moved to collecting just `IpAddr`s in 8fe56d513.